main.py

```diff
@@ -216,11 +218,13 @@ def update(self, circuit_id):
             if enable is False:  # disable if active
                 evc.remove()
             elif path is not None:  # redeploy if active
-                evc.remove()
-                evc.deploy()
+                with evc.lock:
+                    evc.remove()
+                    evc.deploy()
```

Comment:
Here, the first `evc.remove` is not protected by the lock, while the second is.
I'm not sure if `evc.remove` has to be protected by the lock, but both calls should have the same behaviour.

Reply:
Hi @ajoaoff , yes you are right. I think we should also use the lock on evc.remove() because it may free some service tags that could benefit new evcs in a very restricted environment.